flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/A32NX_LocalVarUpdater.js

```diff
@@ -215,5 +232,41 @@ class A32NX_LocalVarUpdater {
         );
     }
 
+    _mfdLcdEffectSelector() {
+        // Calculate opacity for MFD LCD effect
+        const zoomLevel = SimVar.GetSimVarValue('COCKPIT CAMERA ZOOM', 'percent');
+        const camXyz = SimVar.GetGameVarValue('CAMERA POS IN PLANE', 'xyz');
+
+        if (this.zoomLevel !== zoomLevel || this.camZ !== camXyz.z) {
+            // z-axis - away from screen >> -ve
+            // zTarget: Target zPos that marks max pixel effect point [11.625z @ 100%, 11.7z @ 75%]
+            const zTarget = (3975 - zoomLevel) / 333.33;
+            // zΔ: Diff between current zPos and zTarget
+            const zDelta = camXyz.z - zTarget;
+            // opacity: 0 < [4zΔ + 0.5] < 0.5
+            SimVar.SetSimVarValue('L:A32NX_MFD_MASK_OPACITY', 'number', Math.max(0, Math.min(0.5, 4 * (zDelta) + 0.5)));
+
+            this.camZ = camXyz.z;
+            this.zoomLevel = zoomLevel;
+        }
+    }
+
+    _mcduLcdEffectSelector() {
+        // Calculate opacity for MCDU LCD effect
+        const zoomLevel = SimVar.GetSimVarValue('COCKPIT CAMERA ZOOM', 'percent');
+        const camXyz = SimVar.GetGameVarValue('CAMERA POS IN PLANE', 'xyz');
+
+        if (this.zoomLevel !== zoomLevel || this.camY !== camXyz.y) {
+            // y-axis - away from screen >> +ve
+            const yTarget = (zoomLevel + 423.33) / 333.33;
+            const yDelta = yTarget - camXyz.y;
+            // opacity: 0 < [4zΔ + 0.5] < 1
+            SimVar.SetSimVarValue('L:A32NX_MCDU_MASK_OPACITY', 'number', Math.max(0, Math.min(1, 4 * (yDelta) + 0.5)));
+
+            this.camY = camXyz.y;
+            this.zoomLevel = zoomLevel;
+        }
+    }
```

Comment:
These functions should return the value that should be set to the SimVar declared on lines 93 and 99. Due to this not returning the value it tries to set null values, which gives the following console output:

![image](https://user-images.githubusercontent.com/1074786/128378929-b3e0bb1c-eb23-4537-a362-da8021bec93c.png)

Reply:
Resolved in new PR https://github.com/flybywiresim/a32nx/pull/5557